src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,150 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] tables;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        tables = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * tables.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode;
+        int index;
+        if (size > threshold) {
+            resize();
+        }
+        Objects.checkIndex(index = index(key), tables.length);
+        if (key == null) {
+            key = (K) Integer.valueOf(index);
+        }
+        if (tables[index] == null) {
+            newNode = new Node<>(index, key, value, null);
```

Comment:
```suggestion
            newNode = new Node<>(hash, key, value, null);
```
why are you storing index where you need hash?